src/posit/connect/content.py

```diff
@@ -183,6 +183,94 @@ def app_role(self) -> str:
     def id(self) -> str:
         return self.get("id")  # type: ignore
 
+    def update(  # type: ignore
+        self,
+        name: Optional[str] = None,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+        access_type: Optional[str] = None,
+        owner_guid: Optional[str] = None,
+        connection_timeout: Optional[int] = None,
+        read_timeout: Optional[int] = None,
+        init_timeout: Optional[int] = None,
+        idle_timeout: Optional[int] = None,
+        max_processes: Optional[int] = None,
+        min_processes: Optional[int] = None,
+        max_conns_per_process: Optional[int] = None,
+        load_factor: Optional[float] = None,
+        cpu_request: Optional[float] = None,
+        cpu_limit: Optional[float] = None,
+        memory_request: Optional[int] = None,
+        memory_limit: Optional[int] = None,
+        amd_gpu_limit: Optional[int] = None,
+        nvidia_gpu_limit: Optional[int] = None,
+        run_as: Optional[str] = None,
+        run_as_current_user: Optional[bool] = None,
+        default_image_name: Optional[str] = None,
+        default_r_environment_management: Optional[bool] = None,
+        default_py_environment_management: Optional[bool] = None,
+        service_account_name: Optional[str] = None,
+    ) -> None:
+        data: dict[str, Any] = {}
+        if name is not None:
+            data["name"] = name
+        if title is not None:
+            data["title"] = title
+        if description is not None:
+            data["description"] = description
+        if access_type is not None:
+            data["access_type"] = access_type
+        if owner_guid is not None:
+            data["owner_guid"] = owner_guid
+        if connection_timeout is not None:
+            data["connection_timeout"] = connection_timeout
+        if read_timeout is not None:
+            data["read_timeout"] = read_timeout
+        if init_timeout is not None:
+            data["init_timeout"] = init_timeout
+        if idle_timeout is not None:
+            data["idle_timeout"] = idle_timeout
+        if max_processes is not None:
+            data["max_processes"] = max_processes
+        if min_processes is not None:
+            data["min_processes"] = min_processes
+        if max_conns_per_process is not None:
```

Comment:
I think for many of these, we can't check `is not None` because `None` is a valid value in the API, many of these things are int or null. So maybe the check is to see if it is not equal to `self.param`?

Reply:
I figured out a solution using the Ellipsis literal. Working on a full implementation now.

```py
def update(  # type: ignore
        self,
        name: Union[str, EllipsisType] = ...,
        title: Union[Optional[str], EllipsisType] = ...,
    ) -> None:
        kwargs = {}
        if name is not ...:
            kwargs["name"] = name
        if title is not ...:
            kwargs["title"] = title

        response = self.session.patch(self.url, json=kwargs)
        super().update(**response.json())
```

Reply:
I pushed a new commit that uses an override signature to provide the type hinting. The implementation strictly checks if the provided fields are part of the known request body. But we can modify this logic to support forward compatibility.